src/datachain/query/dataset.py

```diff
@@ -352,10 +371,39 @@ def process_udf_outputs(
     udf: "UDFAdapter",
     cb: Callback = DEFAULT_CALLBACK,
     batch_size: int = INSERT_BATCH_SIZE,
+    processed_table: "Table | None" = None,
 ) -> None:
     # Optimization: Compute row types once, rather than for every row.
     udf_col_types = get_col_types(warehouse, udf.output)
 
+    # Track which input sys__ids we've already written to processed table
+    all_processed_sys_ids: set[int] = set()
```

Comment:
How well it will perform in SaaS with huge tables (billions of records)?

How well it will perform with distributed workers since this code is running on a separate node? Does this guarantee uniqueness across all nodes?